src/test/hooks.ts

```diff
@@ -0,0 +1,629 @@
+                  }
+               }
+            })
+         })
+
+         describe('and every non-pre hook throws an error', () => {
+            it('should catch the errors and return the original ctx', async () => {
+               let final_context: any
+               let original_context: any
+
+               const hooks: TinyHooks = {
+                  preSql: (ctx, name, params) => {
+                     original_context = ctx
+
+                     return {
+                        ctx: ctx,
+                        args: [name, params],
+                     }
+                  },
+                  onQuery(_ctx, _query_begin_context) {
+                     throw new Error('Error in onQuery')
+                  },
+                  onSubmit(_ctx, _query_submit_context) {
+                     throw new Error('Error in onSubmit')
+                  },
+                  onResult(_ctx, _query_complete_context) {
+                     final_context = _ctx
+
+                     throw new Error('Error in onResult')
+                  },
+               }
+
+               const hooks_creation_methods = [H.newTiny({ hooks: hooks }), tiny.withHooks(hooks)]
+
+               for (const my_tiny of hooks_creation_methods) {
+                  final_context = null
+
+                  await my_tiny.sql('a.select')
+
+                  expect(final_context).to.deep.equal(original_context)
+               }
+            })
+         })
+
+         describe('and a single non-pre hook throws an error', () => {
+            it('should catch the error and continue with the rest of the hooks', async () => {
+               let final_context: any
+
+               const hooks: TinyHooks = {
+                  preSql: (ctx, name, params) => {
+                     return { ctx: { ...ctx, foo: 'bar' }, args: [name, params] }
+                  },
+                  onQuery(_ctx, _query_begin_context) {
+                     throw new Error('Error in onQuery')
+                  },
+                  onSubmit(ctx, _query_submit_context) {
+                     return {
+                        ...ctx,
+                        onSubmit: 'made it',
+                     }
+                  },
+                  onResult(ctx, query_complete_context) {
+                     final_context = {
+                        ...ctx,
+                        ...query_complete_context,
+                     }
+
+                     return final_context
+                  },
+               }
+
+               const hooks_creation_methods = [H.newTiny({ hooks: hooks }), tiny.withHooks(hooks)]
+
+               for (const my_tiny of hooks_creation_methods) {
+                  final_context = null
+
+                  await my_tiny.sql('a.select')
+
+                  expect(final_context.onSubmit).to.equal('made it')
+               }
+            })
+         })
+      })
+
+      describe('name modification', () => {
```

Comment:
why is it required to return the name at all in the hook when it can't be changed?